apps/src/applab/applab.js

```diff
@@ -1651,3 +1653,18 @@ Applab.updateProperty = function (element, property, value) {
 Applab.isCrosshairAllowed = function () {
   return !Applab.isReadOnlyView && !Applab.isRunning();
 };
+
+Applab.showRateLimitAlert = function () {
+  // only show the alert once per session
+  if (hasSeenRateLimitAlert) {
+    return false;
+  }
+  hasSeenRateLimitAlert = true;
+
+  var alert = <div>The app has been reading or writing to the database too frequently.  Please try again a little later.  Let the developer know if you see this problem again.</div>;
```

Comment:
In past workplaces, not having the string nailed down was more reason to move it to i18n - then it could be changed/fixed without developer intervention.  Not true here?